commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java

```diff
@@ -88,6 +93,42 @@ else if (Validator.isNotNull(engineKey)) {
 		return _commercePaymentMethodGroupRel;
 	}
 
+	public List<CommercePaymentMethodGroupRel>
+			getDefaultCommercePaymentMethodGroupRels()
+		throws PortalException {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)_renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		List<CommercePaymentMethodGroupRel> commercePaymentMethodGroupRels =
+			_commercePaymentMethodGroupRelService.
+				getCommercePaymentMethodGroupRels(
+					themeDisplay.getScopeGroupId());
+
+		Stream<CommercePaymentMethodGroupRel> stream =
+			commercePaymentMethodGroupRels.stream();
+
+		String[] array = stream.map(
+			CommercePaymentMethodGroupRel::getEngineKey
+		).toArray(
+			String[]::new
+		);
+
+		_defaultCommercePaymentMethodGroupRels =
+			addDefaultCommercePaymentMethodGroupRels(
+				_defaultCommercePaymentMethodGroupRels);
+
+		stream = _defaultCommercePaymentMethodGroupRels.stream();
+
+		_defaultCommercePaymentMethodGroupRels = stream.filter(
+			e -> !ArrayUtil.contains(array, e.getEngineKey())
```

Comment:
`e` here should have a more meaningful name -- probably `commercePaymentMethodGroupRel`. See https://docs.google.com/document/d/188YFP0V4n9WJIPYZL3jGBRBvEsvBNmHOU97Fdd_g6Ew/edit#heading=h.a8q4qs756ecb for the general rule.

Reply:
bookmarked, thank you Ethan!